interoptest/src/nodejsservice/src/testservice/service-hopper.js

```diff
@@ -94,6 +116,57 @@ function grpcServiceHop (host, port, request) {
   });
 }
 
+function toBuffer (testRequest) {
+  var bytes = testRequest.serializeBinary();
+  var buffer = new Buffer(bytes);
+  return buffer;
+}
+
+/**
+ * This method sends http request to a server specified
+ * in serviceHop -> request.
+ */
+function httpServiceHop (host, port, request) {
+  console.log(`NextHop HTTP:${host}->${JSON.stringify(request.toObject())}`);
+  const buf = toBuffer(request);
+  const options = {
+    hostname: host,
+    port: port,
+    path: '/test/request',
+    method: 'POST',
+    headers: {
+      'Content-Length': buf.length,
+      'Content-Type': 'application/x-protobuf'
+    }
+  };
+  let response = new interop.TestResponse();
+  return new Promise((resolve, reject) => {
+    try {
+      let req = http.request(options, res => {
+        let data = [];
+        res.on('data', chunk => { data.push(chunk); });
+        res.on('end', () => {
+          if (res.statusCode === 200) {
+            const bytes = new Uint8Array(Buffer.concat(data));
+            response = interop.TestResponse.deserializeBinary(bytes);
+            resolve(response);
+          } else {
+            resolve(setFailureStatus(response, 'Http Service Hopper Error'));
```

Comment:
include statusCode and perhaps error string if available.

Reply:
SGTM, done. 

Examples:
```Http Service Hopper Error: connect ECONNREFUSED 127.0.0.1:10309```
```Http Service Hopper Error 400:Bad Request```
```GRPC Service Hopper Error : 14 UNAVAILABLE: Connect Failed```
```Http Socket Error : Port should be >= 0 and < 65536. Received 103009```